### ISSUE TITLE
TCP Sink block in server mode consumes excessive CPU

### ISSUE THREAD

**Issue by argilo** [gnuradio/gnuradio#6698]
Steps to reproduce:

1. Instantiate a TCP Sink block in server mode.
2. Open a TCP connection to the server.

On my system, this causes the block's CPU utilization to hover at 15% even though the block is idle.

The high CPU utilization is very likely due to this polling loop, which checks every 10 microseconds whether it's time to listen for another connection yet:

https://github.com/gnuradio/gnuradio/blob/c2fc93fdc9448299e9d1004a8744347e020ed9bb/gr-network/lib/tcp_sink_impl.cc#L111-L119

I expect this could be replaced with a condition variable.

**Comment by marcusmueller** [comment]
Correct, still happens. *a LOT* of calls to `nanosleep` in _ZN2gr7network13tcp_sink_impl12run_listenerEv (`gr::network::tcp_sink_impl::run_listener()`). Reproduction, including profiling, in
https://gist.github.com/marcusmueller/6c98d2824755eba115a9326e2bb91244
